implementation/README.md

```diff
@@ -12,32 +16,35 @@ DXE process is very complex; some systems have 750 DXEs.
 
 LinuxBoot replaces most of the UEFI software with Linux. LinuxBoot has an initramfs provided by [u-root](../u-root/README.md).
 
+The above are stored inside a flash filesystem (FFS) inside a region of flash on your motherboard (the BIOS region). Another important region of flash is the ME region. 
+
+The Management Engine (ME) is an x86 CPU embedded in the Intel Platform Controller Hub (PCH). It runs the Minix operating system which boots first and enables hardware such as clocks and GPIOs. ME checks the contents of flash memory and is used to implement "BootGuard". If you reflash and the ME is in "BootGuard" mode, your machine will be unusable. You need to run a tool called `me_cleaner` on the image to disable BootGuard.
+ 
 ## How do you get LinuxBoot on your hardware?
 
 Start with a board running standard UEFI and proceed from "zero changes to FLASH" to "max changes" in 4 steps:
 
 1.  Boot from USB stick via UEFI shell command _or_ netboot (zero changes)
-1.  Replace UEFI Shell code section with Linux kernel (change part of one thing)
-1.  Remove as many DXEs as possible (change by removal). This change:
+2.  Find a way to read flash and write flash
+3.  Understand the flash layout
+4.	Prepare linux kernel and initrd/initramfs payload.
+5. 	Replace UEFI Shell code section with Linux kernel and associated initrd (change part of one thing)
+6.  Remove as many DXEs as possible (change by removal). This change:
 
        *  Speeds boot
        *  Reduces panic possibilities
        *  Removes exploits
        *  In production, it has solved problems
+7.	Clear ME region for initrd storage       
+8.  Replace some DXEs with open source components (change by replacement)
 
-1.  Replace some DXEs with open source components (change by replacement)
+One of the challenges in the above is in finding (or reclaiming) enough space in flash to shoehorn your kernel and initrd into. 
 
 ## Tools of the trade
 
 There are two tools you use when you modify the UEFI flash image: `utk `and `me_cleaner`
```

Comment:
if you are going to put me_cleaner first in the text might as well put it first in this sentence.

Reply:
Which sentence?